Source/WebCore/Modules/permissions/Permissions.cpp

```diff
@@ -203,4 +233,34 @@ void Permissions::query(JSC::Strong<JSC::JSObject> permissionDescriptorValue, DO
         workerLoaderProxy->postTaskToLoader(WTFMove(completionHandler));
 }
 
+#if ENABLE(GEOLOCATION)
+
+static std::optional<PermissionState> determineGeolocationPermissionState(PermissionState permissionState, const Document& document)
+{
+    RefPtr domWindow = document.domWindow();
+    if (!domWindow)
+        return std::nullopt;
+
+    RefPtr geolocation = NavigatorGeolocation::geolocation(domWindow->protectedNavigator());
```

Comment:
This line is causing the crashes. If I don't create a geolocation object here (and don't use it below of course, there's no crash.